prov/verbs/src/verbs_info.c

```diff
@@ -1487,24 +1487,8 @@ static int fi_ibv_handle_ib_ud_addr(const char *node, const char *service,
 {
 	struct ofi_ib_ud_ep_name *dest_addr = NULL;
 	struct ofi_ib_ud_ep_name *src_addr = NULL;
-	void *addr = NULL;
-	size_t len = 0;
 	uint32_t fmt = FI_FORMAT_UNSPEC;
-	int svc = VERBS_IB_UD_NS_ANY_SERVICE, ret = FI_SUCCESS;
-
-	if (node && !ofi_str_toaddr(node, &fmt, &addr, &len) &&
```

Comment:
I think the call to ofi_str_toaddr should remain here to handle string address explicitly formatted for IB UD (i.e. "fi_addr_ib_ud://<gid>:<qpn>:<lid>:<pkey>:<sl>").

Reply:
Are you talking about FI_ADDR_STR or FI_ADDR_IB_UD?

Reply:
"node" as an input is a string, besides being a node name, it can also be interpreted as an address in FI_ADDR_STR format. The case that is handled here is when the string formatted address is a string representation of FI_ADDR_IB_UD, i.e., "fi_addr_ib_ud://......".

Reply:
Got it. So you're saying keep the ofi_str_toaddr call but change the input format to always be FI_ADDR_IB_UD instead of FI_FORMAT_UNSPEC  as it is now?

Reply:
I mean keep the entire block of code as it is now.

Reply:
I see. I missed the call in ofi_str_toaddr to ofi_addr_format where it finds the proper format of the string. That makes more sense now.
In the case that the node is not in the IB UD format and it goes through the FORMAT_UNSPEC path that you added which resolves the node in SOCKADDR format and returns that address/format, it's ok for the getinfo call to still return info->addr_format = FI_ADDR_IB_UD ?

Reply:
Yes it is ok.  I don't expect the change I made to fi_str_toaddr() to affect the behavior here.

Reply:
Alright. Updated it!